### PR TITLE
add acceptance test

### PR DESCRIPTION
**Pull request by voscob**
### Description

Add acceptance test 

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-stripe/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: acceptance_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+				// the method requires NewDestination.
+				"TestSource_Open_ResumeAtPositionSnapshot",
+				// the method requires NewDestination.
+				"TestSource_Open_ResumeAtPositionCDC",
+			},
+			GoleakOptions: []goleak.Option{
+				goleak.IgnoreTopFunction("database/sql.(*DB).connectionOpener"),
+				goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
+				goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+			},
```
Comment (hariso) [inline acceptance_test.go:64]: It would be ideal if we mention why we can't fix these leaks.

File: acceptance_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			SourceConfig:      cfg,
+			DestinationConfig: nil,
+			Skip: []string{
+				// the method requires NewDestination.
+				"TestSource_Read*",
+				// the method requires NewDestination.
+				"TestSource_Open_ResumeAtPositionSnapshot",
+				// the method requires NewDestination.
+				"TestSource_Open_ResumeAtPositionCDC",
```
Comment (hariso) [inline acceptance_test.go:58]: The reason why NewDestination is required, is so we can write something, and then lest the read tests run. If a destination connector is implemented, then we use the destination connector to write data to a test source.

In the absence of a destination connector, what you can do is to implemented the `WriteToSource` function (i.e. you need to implement your own driver, embed ConfigurableAcceptanceTestDriver, and then implement WriteToSource).

Comment (voscob) [reply acceptance_test.go]: Ok, got it.
Done.

File: acceptance_test.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		nameValue        = "client-%s"
+		descriptionValue = "info about the %s"
+	)
+
+	var (
+		name        = fmt.Sprintf(nameValue, uuid.New().String())
+		description = fmt.Sprintf(descriptionValue, name)
+	)
+
+	resource, err := addResource(ctx, cfg, map[string]string{
```
[the remark this answers is not in the document]
Comment (voscob) [reply acceptance_test.go:69]: I have to do it right in this method because I want to know the `id` of added resource to put it to the key.

**Review by hariso — APPROVED** [review]

File: acceptance_test.go [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+		if err != nil {
+			t.Error(err)
+		}
+
+		resource, err := addResource(ctx, cfg, m)
+		if err != nil {
+			t.Error(err)
+		}
+
+		payload, _ := json.Marshal(resource)
```
[the remark this answers is not in the document]
Comment (voscob) [reply acceptance_test.go]: It makes sense.
Added.

File: acceptance_test.go (hunk 1 above)
Comment (voscob) [reply acceptance_test.go]: Deleted.
The reason for the leak was that `HTTP clients' were not closed in tests.

**Pull request merged**